### PR TITLE
fix(udf): unlimit external udf message size

### PR DESCRIPTION
**Pull request by stdrc**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

The previous default 4MB limit by tonic seems too small. After https://github.com/arrow-udf/arrow-udf/pull/62, we can finally increase it to `usize::MAX`.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by stdrc** [comment]
* **#18591** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/18591?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @stdrc and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/expr/impl/src/udf/external.rs [hunk 1]
```diff
@@ -186,7 +190,9 @@ fn get_or_create_flight_client(link: &str) -> Result<Arc<Client>> {
         let client = Arc::new(tokio::task::block_in_place(|| {
             RUNTIME.block_on(async {
                 let channel = connect_tonic(link).await?;
-                Ok(Client::new(channel).await?) as Result<_>
+                let client = FlightServiceClient::new(channel)
+                    .max_decoding_message_size(MAX_DECODING_MESSAGE_SIZE);
```
Comment (fuyufjh) [inline src/expr/impl/src/udf/external.rs:194]: How about making it unlimited? by @BugenZhao

**Review by kwannoel — COMMENTED** [review]
Shall we add a warning trace for udf message with large size? Can be in a separate PR.

**Comment by stdrc** [comment]
> Shall we add a warning trace or metric for udf message with large size? Can be in a separate PR.

It seems not easy to add such log/metric because messages are decoded in tonic and we don't have access to the raw messages. Currently I think we already have `UDF Throughput (bytes)`.

**Review by xxchan — APPROVED** [review]

**Pull request merged**